narwhal/config/src/lib.rs

```diff
@@ -105,11 +105,15 @@ pub type WorkerId = u32;
 /// milliseconds or seconds (e.x 5s, 10ms , 2000ms).
 #[derive(Clone, Debug, Deserialize, Serialize)]
 pub struct Parameters {
-    /// The preferred header size. The primary creates a new header when it has enough parents and
-    /// enough batches' digests to reach `header_size`. Denominated in bytes.
-    pub header_size: usize,
+    /// When the primary has at least `header_num_of_batches_threshold` num of batch digests
+    /// available, then it can propose a new header.
+    pub header_num_of_batches_threshold: usize,
```

Comment:
Since we are adding these parameters, can we give them default values so they don't always have to be set in configs? It will make updating these parameters easier.

Reply:
Good point @mwtian ! I'll update for now those 2 and open a PR to treat the other properties separately - just to make sure things are updated separately